pkg/orchestration/wiring/aws/plugin.go

```diff
@@ -37,12 +37,14 @@ func Resource(resourceType voyager.ResourceType,
 	clusterServicePlanExternalID servicecatalog.PlanExternalID,
 	generateServiceEnvironment ServiceEnvironmentGenerator,
 	envResourcePrefix oap.EnvVarPrefix,
+	optionalShapes func(*orch_v1.StateResource) ([]wiringplugin.ShapeName, error),
```

Comment:
In order to avoid these shenanigans we'd have to create dedicated resource type for SNS, but this means migrating existing SDs that define any SNS.

We could do that, but 1. as a separate step. 2. It might take time to deprecate old resource.